src/scripts/main.js

```diff
@@ -1,3 +1,224 @@
+  const body = document.body;
+  const form = document.createElement('form');
+
+  form.classList.add('new-employee-form');
+
+  for (const title of HEADERS) {
+    const label = document.createElement('label');
+
+    label.innerText = `${title[0].toUpperCase() + title.slice(1)}:`;
+
+    if (title === 'office') {
+      const selectBox = document.createElement('select');
+
+      selectBox.name = title;
+      selectBox.setAttribute('data-qa', title);
+      selectBox.required = true;
+
+      OFFICES.forEach((office) => {
+        const option = document.createElement('option');
+
+        option.value = office;
+        option.innerText = office;
+        selectBox.appendChild(option);
+      });
+
+      label.appendChild(selectBox);
+      form.appendChild(label);
+      continue;
+    }
+
+    const input = document.createElement('input');
+
+    input.name = title;
+    input.required = true;
+
+    if (title === 'age' || title === 'salary') {
+      input.type = 'number';
+      input.min = 0;
+    } else {
+      input.type = 'text';
+    }
+
+    label.appendChild(input);
+    form.appendChild(label);
+  }
+
+  const button = document.createElement('button');
+
+  button.type = 'submit';
+  button.innerText = 'Save to table';
+
+  form.appendChild(button);
+  body.appendChild(form);
+}
+
+addForm();
+
+const pushNotification = (title, description, type) => {
+  const table = document.querySelector('table');
+
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+  notification.className = `notification ${type}`;
+
+  notification.innerHTML = `
+    <h2 class="title">${title}</h2>
+    <p>${description}</p>
+  `;
+  table.insertAdjacentElement('afterend', notification);
+
+  setTimeout(() => {
+    notification.style.display = 'none';
+  }, 2000);
+};
+
+if (document.querySelector('.new-employee-form')) {
+  const form = document.querySelector('.new-employee-form');
+
+  form.addEventListener('submit', (e) => {
+    e.preventDefault();
+
+    const data = new FormData(form);
+    let isValid = true;
+    const newName = data.get('name');
+    const age = data.get('age');
+    const salary = parseInt(data.get('salary')).toLocaleString('en-US');
+
+    if (newName.length < 4) {
+      isValid = false;
+      pushNotification('Oops!', 'Value has less than 4 letters', 'error');
+    }
+
+    if (+age <= 18 || +age > 90) {
+      isValid = false;
+
+      pushNotification(
+        'Oops!',
+        'Value is less than 18 or more than 90',
+        'error',
+      );
+    }
+
+    if (isValid) {
+      const newEmployee = {
+        name: newName,
+        position: data.get('position'),
+        office: data.get('office'),
+        age,
+        salary: '$' + salary,
+      };
+
+      items.push(newEmployee);
+      rebuildList(items);
+
+      pushNotification(
+        'Success!',
+        'New employee has added to list.',
+        'success',
+      );
+    }
+  });
+}
```

Comment:
The form submission handling logic can be broken down into smaller functions. For example, validation can be a separate function, creating a new employee object can be another function, and so on. This will make the code easier to understand and maintain.